src/map-server.ts

```diff
@@ -0,0 +1,145 @@
+   *
+   * @param importId The ID for the desired import.
+   * @returns The import record, or undefined if no import with the given ID exists.
+   */
+  getImport(importId: string): undefined | ImportRecord {
+    return this.#api.getImport(importId)
+  }
+
+  listen(
+    ...args: Parameters<FastifyInstance['listen']>
+  ): ReturnType<FastifyInstance['listen']> {
+    return this.fastifyInstance.listen(...args)
+  }
+
+  async close() {
+    if (this.#activeImports.size > 0) {
+      // Wait for all worker threads to finish, so we don't terminate the thread
+      // without closing the DB connections in thread. This is kind-of hacky:
+      // It relies on the MessagePort being closed when the worker thread is done.
+      await Promise.all(
+        [...this.#activeImports.values()].map(
+          (port) =>
+            new Promise((res) => {
+              port.once('close', res)
+              port.once('error', res)
+            })
+        )
+      )
+    }
+
+    await this.#piscina.destroy()
+
+    this.#db.close()
+
+    this.fastifyInstance.close()
+  }
+}
+
+function setUpDatabase(db: Database) {
+  // Enable auto-vacuum by setting it to incremental mode
+  // This has to be set before the anything on the db instance is called!
+  // https://www.sqlite.org/pragma.html#pragma_auto_vacuum
+  // https://www.sqlite.org/pragma.html#pragma_incremental_vacuum
+  db.pragma('auto_vacuum = INCREMENTAL')
+
+  // Enable WAL for potential performance benefits
+  // https://github.com/JoshuaWise/better-sqlite3/blob/master/docs/performance.md
+  db.pragma('journal_mode = WAL')
+
+  migrate(db, path.resolve(__dirname, '../prisma/migrations'))
+
+  // Any import with an `active` state on startup most likely failed due to the server process stopping
+  // so we update these import records to have an error state
+  convertActiveImportsToErrorImports(db)
+}
+
+function createPiscina() {
+  const result = new Piscina({
+    filename: path.resolve(__dirname, './lib/mbtiles_import_worker.js'),
+    minThreads: 1,
+  })
+
+  result.on('error', (error) => {
+    // TODO: Do something with this error https://github.com/piscinajs/piscina#event-error
+    console.error(error)
+  })
+
+  return result
+}
+
+function createMapFastifyServer(
+  api: Api,
+  fastifyOpts: FastifyServerOptions
+): FastifyInstance {
+  const result = createFastify(fastifyOpts)
+
+  result.decorate('api', api)
+
+  result.register(FastifyStatic, {
+    root: SDF_STATIC_DIR,
+    prefix: '/fonts',
+    // res type documented in @fastify/static@5 docs is misleading
+    setHeaders: (res: any, path: string) => {
+      if (path.toLowerCase().endsWith('.pbf')) {
+        res.setHeader('Content-Type', 'application/x-protobuf')
+      }
+    },
+  })
+
+  for (const name of Object.keys(routes) as Array<keyof typeof routes>) {
+    result.register(routes[name], { prefix: '/' + name })
+  }
+
+  return result
+}
```

Comment:
All of this was moved from the old creation of the Fastify server.

The main difference: `Api` is no longer a Fastify plugin so we just call `.decorate('api', api)`.